apps/wallet/src/ui/app/pages/dapp-tx-approval/TransactionSummaryCard.tsx

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { MiniNFT } from './MiniNFT';
+import {
+    SummaryCard,
+    SummaryCardHeader,
+    SummaryCardContent,
+} from './SummaryCard';
+import AccountAddress from '_components/account-address';
+import ExplorerLink from '_components/explorer-link';
+import { ExplorerLinkType } from '_components/explorer-link/ExplorerLinkType';
+import ExternalLink from '_components/external-link';
+import { useMiddleEllipsis, useFormatCoin, useGetNFTMeta } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+import type { CoinsMetaProps } from '_redux/slices/transaction-requests';
+
+import st from './DappTxApprovalPage.module.scss';
+
+const TRUNCATE_MAX_LENGTH = 10;
+const TRUNCATE_PREFIX_LENGTH = 6;
+
+type TransferSummerCardProps = {
+    coinsMeta: CoinsMetaProps[];
+    origin: string;
+    objectIDs: string[];
+    gasEstimate: number | null;
+};
+
+function MiniNFTLink({ id }: { id: string }) {
+    const objectId = useMiddleEllipsis(
+        id,
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+    const nftMeta = useGetNFTMeta(id);
+    return (
+        <>
+            {nftMeta && (
+                <MiniNFT
+                    url={nftMeta.url}
+                    name={nftMeta?.name || 'NFT Image'}
+                    size="xs"
+                />
+            )}
+            <ExplorerLink
+                type={ExplorerLinkType.object}
+                objectID={id}
+                className={st.objectId}
+                showIcon={false}
+            >
+                {objectId}
+            </ExplorerLink>
+        </>
+    );
+}
+
+function CoinMeta({
+    receiverAddress,
+    coinMeta,
+    origin,
+}: {
+    receiverAddress: string;
+    coinMeta: CoinsMetaProps;
+    origin: string;
+}) {
+    const [formattedAmount, symbol] = useFormatCoin(
+        coinMeta.amount ? Math.abs(coinMeta.amount) : 0,
+        coinMeta.coinType
+    );
+
+    /// A net positive amount means the user received coins and verse versa.
+    const sendLabel = coinMeta.amount < 0 ? 'Send' : 'Receive';
+    const receiveLabel = coinMeta.amount < 0 ? 'To' : 'From';
+
+    return (
+        <div className={st.content}>
+            <div className={st.row}>
+                <div className={st.label}>{sendLabel}</div>
+                <div className={st.value}>
+                    {formattedAmount} {symbol}
+                </div>
+            </div>
+
+            <div className={st.row}>
+                <div className={st.label}>{receiveLabel}</div>
+                <div className={st.value}>
+                    <div className={st.value}>
+                        <ExternalLink
+                            href={origin}
+                            className={st.origin}
+                            showIcon={false}
+                        >
+                            {new URL(origin).host}
```

Comment:
Why do we consider the host to be the receiver of a coin? Isn't there an actual receiver address we can use?

Reply:
For most of the dry_run txns, the receiver and sender/user addresses are the same.  if they are I will use the origin.

Reply:
IMO we actually should _never_ use the origin, it's super misleading. If the address is the same then we should figure out why, and address that.